src/ragger/backend/speculos.py

```diff
@@ -139,3 +141,22 @@ def save_screen_snapshot(self, path: Path) -> None:
         screenshot = self._client.get_screenshot()
         snap = Image.open(BytesIO(screenshot))
         snap.save(path)
+
+    def wait_for_screen_change(self, timeout: float = 10.0, context: list = []):
+        start = time()
+        if not context:
+            return self._client.get_screen_content()
+        else:
+            content = context
+            while content == context:
+                content = self._client.get_screen_content()
+                now = time()
+                if (now - start > timeout):
+                    raise TimeoutError("Timeout waiting for screen change")
```

Comment:
Add a `time.sleep(x)` here to give some time to other threads?

Reply:
I don't understand what would be the purpose of the sleep

Reply:
sorry, I put in on the wrong line.
I mean putting a sleep in the while loop to avoid polling `get_screen_content()` to fast.
I guess 0.1s between each poll won't slow tests?